jgrapht-core/src/main/java/org/jgrapht/graph/DirectedAcyclicGraph.java

```diff
@@ -124,7 +123,12 @@ protected DirectedAcyclicGraph(
         VisitedStrategyFactory visitedStrategyFactory, TopoOrderMap<V> topoOrderMap,
         boolean weighted)
     {
-        super(vertexSupplier, edgeSupplier, weighted);
+        super(
+            vertexSupplier, edgeSupplier,
+            new DefaultGraphType.Builder()
+                .directed().allowMultipleEdges(true).allowSelfLoops(false).weighted(weighted)
```

Comment:
Shouldn't we allow the caller to determine whether multiple edges are allowed (with the default being false for backwards compatibility)?

Reply:
Any ideas how to do that? Should we add another constructor with a `multipleEdges` parameter? Or  should we add a constructor with a `GraphType` parameter?

Reply:
Added backwards compatibility as you suggested.